google/cloud/security/enforcer/gce_firewall_enforcer.py

```diff
@@ -486,6 +486,9 @@ def add_rules(self, rules, network_name=None):
           InvalidFirewallRuleError: One or more rules failed validation.
         """
         for rule in rules:
```

Comment:
I believe the check should be before this for loop, i.e. make sure we are getting a list of rules.